databases/week2/queries.sql

```diff
@@ -0,0 +1,173 @@
+  id = 36;
+--Change a task due date
+UPDATE
+  task
+SET
+  due_date = '2021-08-06''13:01:25'
+WHERE
+  id = 36;
+--Change a task status
+UPDATE
+  task
+SET
+  status_id = 2
+WHERE
+  id = 36;
+--Mark a task as complete
+UPDATE
+  task
+SET
+  status_id = 3
+WHERE
+  id = 36;
+--Delete a task
+DELETE FROM
+  task
+WHERE
+  id = 36;
+--Part 2: School database
+  --Create a new database containing the following tables:
+  --Class: with the columns: id, name, begins (date), ends (date)
+  --Student: with the columns: id, name, email, phone, class_id (foreign key)
+  CREATE database school_db;
+USE school_db;
+CREATE Table `class`(
+    `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` VARCHAR(255) NOT NULL,
+    `begins(date)` DATE,
+    `ends(date)` DATE
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+select
+  *
+from
+  class;
+CREATE Table `student`(
+    `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` VARCHAR(255) NOT NULL,
+    `email` VARCHAR(255) NOT NULL,
+    `phone` VARCHAR(255) NULL,
+    `class_id` int(10) UNSIGNED,
+    CONSTRAINT `fk_class` FOREIGN KEY (`class_id`) REFERENCES `class`(`id`) ON DELETE CASCADE ON UPDATE CASCADE
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+--Create an index on the name column of the student table.
+  CREATE index idx_name ON student(name);
+SHOW INDEX
+FROM
+  student;
+--Add a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).
+  USE school_db;
+ALTER table
+  class
+ADD
+  COLUMN `status` enum('not-started', 'ongoing', 'finished') NOT NULL;
+--Part 3: More queries
+  --You should do these queries on the database hyf_lesson2, which we created last class.
+  --Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT
+  user_id,
+  task_id,
+  title,
+  user.email
+FROM
+  user_task
+  JOIN task ON task_id = task.id
+  JOIN user ON user_id = user.id
+WHERE
+  user.email LIKE '%@spotify.com';
+--Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT
+  task.title,
+  user.name AS user_name,
+  status.name AS status_name
+FROM
+  task
+  RIGHT JOIN status ON task.status_id = status.id
+  RIGHT JOIN user_task ON task.id = task_id
+  JOIN user ON user_task.user_id = user.id
+WHERE
+  user.name = 'Donald Duck'
+  and status.name = 'Not started';
+--Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT
+  task.title,
+  user.name AS user_name,
+  task.created
+FROM
+  task
+  JOIN user_task ON task.id = task_id
+  JOIN user ON user_task.user_id = user.id
+WHERE
+  user.name = 'Maryrose Meadows'
+  and month(created) = 9;
+--Find how many tasks where created in each month, e.g. how many tasks were created in october, how many tasks were created in november, etc. (hint: use group by)
+SELECT
+  count(*),
+  month(created)
```

Comment:
If you give this a name, e.g. month(created) as MonthCreated you can use that in your group by

Reply:
Yes- I have given a name and used it in group by as well